velox/exec/Driver.h

```diff
@@ -532,6 +570,25 @@ class Driver : public std::enable_shared_from_this<Driver> {
 
   void updateStats();
 
+  // Defines the driver barrier processing state.
+  struct BarrierState {
+    // If set, the driver has started output draining. It points to the operator
+    // that is currently draining output.
+    std::optional<int32_t> drainOpId{std::nullopt};
+    // If set, the specified operator doesn't need any more input to finish the
+    // draining operation. All the upstream operators within the same driver
+    // should suppress their output or output processing.
+    std::optional<int32_t> suppressInputOpId{std::nullopt};
+  };
+
+  // Invoked to start draining on the next operator.
+  void startNextDrainOp();
```

Comment:
nit: drainNextOperator or startDrainingNextOperator.

Perhaps, clarify what happens if there is no "next" operator and that "next" refers to the immediate downstream operator.